accepted/00000-hub-reporting.md

```diff
@@ -0,0 +1,199 @@
+  stay empty. Columns only available on the Uyuni Server will not be exported.
+
+Implementing this as a taskomatic QueueJob could be an option. The Queue Job is started and collect a list
+of candidates. A number of parallel workers can be specified to connect to every single server instance.
+
+## Show data on reporting tool
+In order to use the data stored in the hub and/or in the servers as a datasource for external reporting tools,
+the admin should create a read-only user to query information from DBs.
+
+# Documentation
+Documentation will cover:
+- reporting tool architecture
+- user creation/management
+- ...
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+
+# Alternatives
+[alternatives]: #alternatives
+
+- SQL or NoSQL DB?
+  - Reporting tools often support PostgreSQL directly why we could not find a similar good support
+    for NoSQL databases
+  - very limited knowledge in NoSQL makes it hard to decide for a technology we do not know
+  - use case seems better to fit of SQL (based on a very limited knowledge about NoSQL databases)
+  - no obvious killer features provided by NoSQL which we want to use
+
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Q: Should we handle the scenario of multiple hierarchical hubs?
+  * Q1: Is this setup allowed and, if so, how is it handled by the existing APIs and features?
+    - A1: it is not forbidden, but the XMLRPC API forwarder will not work with such a setup.
+  * Q1: How do we handle and translate the server ids?
+     - We can internally remap all the ids in the hub database to make sure they are unique
+     - We need to keep track of the 'server path' to make sure the end-user is able to identify the
+       client within the hierarchical network.
+
+- Q: Is implementation in Java with Hibernate possible and fast enough for
+  * Q1: filling the report table in a single Server?
+  * Q2: to collect the data from multiple single Servers and insert them into the Hub DB?
+
+- Q: should we use the same tooling for the DB schema as we use for the main Uyuni Schema?
+  - A: yes, no reason for introducing something new and be different from what we already know.
+
+- Q: where and how to configure the connection parameters for the reporting database
+  * Q1: on a single Server ?
+    - A1: The R/W account should be in `/etc/rhn/rhn.conf` like the other DB account
+    - A1: additional accounts should be stored where they are needed. The single server only needs the R/W account. Users which want to connect a reporting tool to it, should remember it or store it in the account management of the reporting tool.
+  * Q2: on the Hub for every managed single Server?
+    - A2: As the Hub "manages" the single Servers, it should create an own Read-only account and store it in its DB under the system item.
+
+- Q: how to identify a Uyuni Server in the Hub database and distinguish it from a normal managed client?
+  - (?): current assumption in Hub is, that all systems are Uyuni Servers.
```

Comment:
This is a correct assumption. Although, I can't remember if we already have some rules in place which enforce it.

Reply:
Richardo told me that. We are trying to find an easy test to officially mark a system as a Uyuni/SUSE Manager Server.
Currenly I think about:
- checking for a special file
- connecting to the XMLRPC API on "localhost"
More ideas are welcome.

Reply:
Can installed products data be used to this end?

Reply:
Maybe we can see if the pattern is installed?

Reply:
I was thinking about future container work. What is a SUSE Manager or Uyuni Server when it comes to containers.
I think the central part is the Java Public API (XMLRPC or HTTP). So I was thinking about a connection check to localhost.
But this is not that easy.

Reply:
Differently from Proxy's case, I do not see Servers being something other than managed minions in the immediate future. That will mean, one of the containers will have a salt-minion instance running.

Long term we might want to remove it as we are doing with the Proxy, but that is not something I see coming for at least the first version.